### PR TITLE
Kill pathfinder

### PR DESCRIPTION
**Pull request by csasarak**
# Overview

This PR:

* Kills the pathfinder binary.
* Moves the `LicenseAnalyzeProject` into its own module. This typeclass is for declared license scanning something which I think we've discussed wanting in the future, so I didn't delete.

I've put these changes in an `Unreleased` block in our readme because while this is a user-visible change I don't think that it warrants a new release. I can change this if a reviewer disagrees since we generally try to release things ASAP.

## Acceptance criteria

Pathfinder is gone.

## Risks

A customer somewhere was using `pathfinder` and we didn't know. It hasn't been updated for a very long time though so they can still get it from an old release if this is the case.

## References

- [ANE-1239](https://fossa.atlassian.net/browse/ANE-1239): Kill Pathfinder

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this PR added docs, I added links as appropriate to the user manual's ToC in `docs/README.ms` and gave consideration to how discoverable or not my documentation is.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json` AND I have updated example files used by `fossa init` command. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

[ANE-1239]: https://fossa.atlassian.net/browse/ANE-1239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/App/Fossa/Analyze/LicenseAnalyze.hs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module App.Pathfinder.Types (
+module App.Fossa.Analyze.LicenseAnalyze (
```
Comment (JeffreyHuynh1) [inline src/App/Fossa/Analyze/LicenseAnalyze.hs:1]: Do you think it's worthwhile to write a comment here as to why we're keeping `LicenseAnalyzeProject` and moving it into its own module?

**Comment by JeffreyHuynh1** [comment]
All the changes here look good! 

I was a little curious about PathFinder though because I've never worked with it. What was the intended purpose of PathFinder and why is it that we're deleting it now? I understand that there hasn't been any recent work done with PathFinder, but aside from that, that's the only thing I know about it.

**Review by JeffreyHuynh1 — APPROVED** [review]

File: src/App/Fossa/Analyze/LicenseAnalyze.hs (hunk 1 above)
Comment (csasarak) [reply src/App/Fossa/Analyze/LicenseAnalyze.hs:8]: Good question, and yes. I've done this.

**Comment by csasarak** [comment]
> I was a little curious about PathFinder though because I've never worked with it. What was the intended purpose of PathFinder and why is it that we're deleting it now?

It was an executable that would check manifest files (like `Cargo.toml`, `pom.xml`, etc.) for declared licenses. I'm not exactly sure what the history of it being a separate binary and not a part of regular `fossa` is.

**Pull request merged**